modules/http4s/src/smithy4s/http4s/EndpointSpecificMiddleware.scala

```diff
@@ -0,0 +1,38 @@
+package smithy4s
+package http4s
+
+import org.http4s.HttpApp
+
+// format: off
+trait EndpointSpecificMiddleware[F[_]] {
+  def prepare[Alg[_[_, _, _, _, _]]](service: Service[Alg])(
+      endpoint: Endpoint[service.Operation, _, _, _, _, _]
```

Comment:
I like this so much.

also, is there an argument for why Endpoint couldn't have IEOSESO as type members?

Reply:
Hmm I'm not sure it'd be possible because the same middleware implementation will be used across multiple endpoints in a service so the IOESISO will change depending on the endpoint

Reply:
no I mean they'd be type members _in_ `Endpoint`. I think it might just be an inconvenience because it makes `Transformation[Endpoint[Op, ...]]` impossible to express directly

Reply:
Oh I see, sorry I misread your comment. Yeah I'm not sure about that one tbh

Reply:
It may be achievable, but it should be its own PR. Want to have a go at it @kubukoz ?

Reply:
perhaps